src/app/obzboard-set.ts

```diff
@@ -106,6 +108,53 @@ export class OBZBoardSet implements ImageResolver, SoundResolver {
     return observables.length > 0 ? forkJoin(...observables) : of(true);
   }
 
+  private blobifySounds(httpClient: HttpClient, progress: ProgressService): Observable<boolean> {
+
+    const observables = [];
+
+    this.boards.forEach((board: OBFBoard) => {
+      observables.push(this.progressObservable('Pre-caching sounds.', progress));
+      board.sounds.forEach((sound: Sound) => {
+        // if sound already has a path, then don't worry
+        if (!sound.path) {
+          // try data first
+          if (sound.data && sound.contentType) {
+            const key = `data:${sound.id}`;
+            // TODO: we could use the content type from here that we're stripping off
+            const data = sound.data.substr(sound.data.indexOf(',') + 1);
+            this.setSound(key, data);
+            // set path to key so render code will look up blob in sounds
+            sound.path = key;
+            // remove data as we don't need to cache it too
+            sound.data = null;
+          } else if (sound.url) {
+            // load url into blob
+            console.log(`Loading sound ${sound.url}`);
+            observables.push(httpClient.get(sound.url, { responseType: 'blob' }).pipe(
+              mergeMap((blob: Blob): Observable<Event> => {
+                const reader = new FileReader();
```

Comment:
Probably worth raising a follow on ticket for the error handling
(maybe something like https://gist.github.com/iansinnott/3d0ba1e9edc3e6967bc51da7020926b0)

Reply:
Done

Reply:
Issue #154